### DISCUSSION TITLE
Is neovim 0.7+ really required

### DISCUSSION THREAD

**Discussion by danielo515** [mrjones2014/legendary.nvim#109] [General]
Hello, this plugin looks great.
I looked at the features it adds, and I don't see anything that is not available on neovim 0.6.1 
Is neovim 0.7 really required? I want to try this plugin a lot, but neovim 0.7 was breaking my lunarvim install, so I can't use it.

Regards

**Comment by mrjones2014** [comment]
Yes, it uses the APIs listed here: https://github.com/mrjones2014/legendary.nvim#prerequisites

These are only available in Neovim 0.7+

EDIT: technically you could use legendary.nvim on Neovim 0.6.1 as long as you don't have it actually _bind_ your keymaps, and only use it for the finder.

**Reply by mrjones2014** [reply]
I suggest filing an issue with LunarVim, they'll need to get it working for Neovim 0.7+ at some point anyway. Neovim 0.7 will be stabilizing and released as a stable release within a few weeks.

**Reply by danielo515** [reply]
So let's say I already have my key maps defined as a table, including which key bindings (which is the  ase for lunar vim anyway). Will it be possible to pass them to this plugin as a configuration option and just let it search and call them? Will I need to forke it to remove the security checks?

**Reply by mrjones2014** [reply]
I believe the way it's working right now is you can pass which-key tables in but they'll still be bound by legendary.

Does which-key actually _bind/set_ they keymaps? If so I can just make it so that legendary doesn't attempt to bind them again, and you can pass you which-key tables in as-is.

**Reply by danielo515** [reply]
I guess they do? But I can't say for sure. I know it is possible to add them at runtime using a method that wich key exposes, but that is something I guess it is just vim built in? I'm not sure without looking at their code

**Reply by mrjones2014** [reply]
Can you test your which-key config on this branch and let me know if it works?

https://github.com/mrjones2014/legendary.nvim/pull/111

Easiest way is just to call `require('legendary').setup()` before ever calling `require('which-key').register(tables, opts)`

**Reply by mrjones2014** [reply]
@danielo515 were you able to test that branch?

**Reply by danielo515** [reply]
Tried to install from that branch, but doesn't seem to exist anymore

**Reply by mrjones2014** [reply]
I closed the PR, as I hadn't heard anything and it's a bit out of date now anyway. I'll work on getting a new one up.

**Reply by mrjones2014** [reply]
@danielo515 try #122

**Reply by danielo515** [reply]
tried branch 122, but I still get that legendary requires lvim version 0.7.0 or higer, which I'm kid of tempted to try again since someone told me that neovim 7 is working fine with lunar vim

**Reply by mrjones2014** [reply]
Oh, I forgot, I have explicit checks in there. I need to move those checks into a different place for it to work. Give me a few mins.

**Reply by mrjones2014** [reply]
@danielo515  try now.

**Reply by danielo515** [reply]
I tried the latest version (with a commit which is about 70 mintues old) and I still get the same error message if I use neovim 0.6.1
But don't bother anymore, I am moving to neovim 0.7, since it is working nice with my curent setup

**Comment by danielo515** [comment]
No, not yet, sorry.
I will give it a try this evening

On Fri, Apr 8, 2022 at 1:15 PM Mat Jones ***@***.***> wrote:

> @danielo515 <https://github.com/danielo515> were you able to test that
> branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrjones2014/legendary.nvim/discussions/109#discussioncomment-2530243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARKJWP35E7YVBET4GJZAODVEAIODANCNFSM5SVIB5BA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
---
https://danielorodriguez.com
